src/js/popup.js

```diff
@@ -469,25 +469,34 @@ $(document).ready(function () {
                     Materialize.toast('Error reading file. Please try again', 2000, 'toastNotification');
                 };
                 reader.onload = function () {
-                    var data = $.parseJSON(reader.result);
-                    chrome.runtime.sendMessage({
-                        JDTIfunction: 'upload',
-                        fileContents: data
-                    }, function (response) {
-                        if (response.status === 'success') {
-                            location.reload();
-                            Materialize.toast('Templates successfully loaded from file', 2000, 'toastNotification');
-                            dmTemplateUpdate('upload');
-                        } else {
-                            if (response.message) {
-                                dmError('upload', response.message);
-                                Materialize.toast(response.message, 2000, 'toastNotification');
+                    let validJSON = true;
+                    try {
+                        var data = JSON.parse(reader.result);
+                    } catch (err) {
+                        validJSON = false;
+                        dmError('upload', 'Error Parsing JSON');
+                        Materialize.toast('Error Parsing JSON', 2000, 'toastNotification');
+                    }
+                    if (validJSON) {
+                        chrome.runtime.sendMessage({
+                            JDTIfunction: 'upload',
+                            fileContents: data
+                        }, function (response) {
+                            if (response.status === 'success') {
+                                location.reload();
+                                Materialize.toast('Templates successfully loaded from file', 2000, 'toastNotification');
+                                dmTemplateUpdate('upload');
                             } else {
-                                dmError('upload', 'generic');
-                                Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                                if (response.message) {
+                                    dmError('upload', response.message);
+                                    Materialize.toast(response.message, 2000, 'toastNotification');
```

Comment:
I'm assuming https://github.com/rdbrck/jira-template-injector/pull/41/files#diff-9566d9c6e04bce6d87d5cbfbf1062b04L449 erroring was the reason nothing was happening for both the upload and download buttons and the reason this wasn't showing anything?

Reply:
Yes. I wasn't checking that domains existed. Now fixed.